odl/discr/discr_mappings.py

```diff
@@ -467,219 +482,307 @@ def _call(self, x, out=None):
             ``out`` was provided, the returned object is a reference
             to it.
         """
+        # TODO: pass reasonable options on to the interpolator
         def linear(arg, out=None):
             """Interpolating function with vectorization."""
             if is_valid_input_meshgrid(arg, self.grid.ndim):
-                # TODO: check if this works for 'F' ordering
-                interp = _LinearMeshgridInterpolator(
+                interp = _LinearInterpolator(
                     self.grid.coord_vectors,
-                    x.data.reshape(self.grid.shape, order=self.order))
+                    x.data.reshape(self.grid.shape, order=self.order),
+                    typ='meshgrid')
```

Comment:
Or `in_type`. I think I saw `typ` as a replacement for `type` as argument somewhere. Not so great, yeah.